### PR TITLE
Node.js 4.0.0 upgrade and code fixes

### PR DESCRIPTION
**Pull request by maria-robobug**
- `QueryOptions` should be `TransactionQueryOptions` in `transactions-example.ts`
- Driveby dependency update for Ottoman (since I'm editing the package.json I might as well)
- Minor fixes and test adjustments due to Node.js 4.0 upgrade

### CODE REVIEW

File: modules/test/test-howtos.bats [hunk 1]
```diff
@@ -58,17 +58,22 @@ load 'test/test_helper.bash'
 
 @test "[howtos] - search.js" {
     runExample $HOWTOS_DIR search.js
-    assert_success
+    # This whole example is going to fail due to the below bug:
+    # https://issues.couchbase.com/browse/JSCBC-1044
+    assert_failure
+
+    # TODO: Uncomment (lines 66-76) when JSCBC-1044 is fixed
```
Comment (osfameron) [inline modules/test/test-howtos.bats]: might be worth adding that as diagnostic output, so we continue to see it?

I use `diag` https://github.com/couchbaselabs/try-cb-test/blob/main/test-helpers.bash#L66-L70
which makes sure it's commented correctly.

File: modules/howtos/examples/subdoc.js [hunk 2]
```diff
@@ -152,13 +152,23 @@ async function go() {
...
 
   console.log('mutate-createpaths')
-  // tag::mutate-createpaths[]
-  await collection.mutateIn('customer123', [
-    couchbase.MutateInSpec.arrayAppend('some.path', 'Hello World', {
-      createPath: true,
-    }),
-  ])
-  // end::mutate-createpaths[]
+  try {
```
Comment (osfameron) [inline modules/howtos/examples/subdoc.js:155]: this commit looks fine, though the subject looks more like "deal with JSCBC-1045" than Minor Fixes?

File: modules/howtos/examples/views.js [hunk 3]
```diff
@@ -19,12 +22,45 @@ async function go() {
...
-  var result = bucket.viewQuery('landmarks', 'by_name', {
-    key: 'statue-of-liberty',
+  var result = await bucket.viewQuery('landmarks', 'by_name', {
+    key: 'landmark_10019',
   })
-
   // end::key-query[]
 }
+
+async function createViews(bucket) {
```
Comment (osfameron) [inline modules/howtos/examples/views.js:31]: This also doesn't look like a minor fix.
How did this work before?

File: modules/howtos/examples/user-manager.js [hunk 4]
```diff
@@ -39,6 +39,10 @@ async function example1(clusterAdm, testUsername, testPassword, bucketName) {
         ]
     })
     // end::usermanagement_1[]
+
+    // Give some time (2 seconds) for user to be created as example code
+    // might run too quickly.
+    await new Promise((resolve) => setTimeout(resolve, 2000))
```
Comment (osfameron) [inline modules/howtos/examples/user-manager.js:45]: Is there a JSCBC report for these 2 new slowups?
Race condition in testing doesn't feel great.

File: modules/test/test-howtos.bats (hunk 1 above)
Comment (maria-robobug) [reply modules/test/test-howtos.bats]: Yeah useful output, let me add.

**Review by osfameron — APPROVED** [review]
👍 
few questions/nitpicks inline

File: modules/howtos/examples/views.js (hunk 3 above)
Comment (maria-robobug) [reply modules/howtos/examples/views.js:31]: It just ran happily but didn't actually work it seems (the views referenced don't exist). I created the views so the code can execute happily, but didn't want to spend too much time on this as supposedly Views will go away 🤞 .

File: modules/howtos/examples/user-manager.js (hunk 4 above)
Comment (maria-robobug) [reply modules/howtos/examples/user-manager.js:45]: No, maybe worth asking at the Friday meeting this, as they probably would have performance tested the SDK? 
Could be more a "hey we noticed this took a fraction longer, maybe you've seen this as well?"

**Comment by maria-robobug** [comment]
Bizarre, Ottoman example keeps failing, doesn't happen locally though 😕  - I'll dig into it

File: modules/howtos/examples/user-manager.js (hunk 4 above)
Comment (osfameron) [reply modules/howtos/examples/user-manager.js:45]: Yeah, I guess *quantitative* slowups isn't any of our business...

BUT: immediate -> takes time, is a qualitative change isn't it?
It feels a bit like a new behaviour change (as we have to add poll/race-sleep to workaround)

**Comment by maria-robobug** [comment]
> oh, yes, so this line came from the try-cb-test.
I can't remember exact details, but I think the echo didn't work, so I left it commented there to remind me to look into it, and to note why I'm using the less common printf instead.

Possibly `echo` gets swallowed up by bats? I'll remove that commented out bit to avoid confusion and merge.
I'll investigate the Ottoman test failure in a separate PR 🤔 .

**Comment by maria-robobug** [comment]
>_< ok, Ottoman test now fine, I'll keep an eye on this in case it happens again in future...

**Pull request merged**